src/erlcloud_retry.erl

```diff
@@ -70,8 +70,18 @@ request_and_retry(Config, ResultFun, {retry, Request}, MaxAttempts) ->
         erlcloud_aws:get_timeout(Config), Config),
     case Rsp of
         {ok, {{Status, StatusLine}, ResponseHeaders, ResponseBody}} ->
+            ResponseType = case Status >= 200 andalso Status < 300 of
```

Comment:
as much as it is the only place to put it here but we should not pollute general retry module with Lambda specifics.
but rather create a dedicate retry function for lambda and use it.

Reply:
one the core reasons - this  will retry it silently but with will lose the ability to even log what went wrong.
was it a function error of USER or AWS lambda function error

Reply:
The retry function wouldn't be  called in case ResponseType is 'ok'.
I can generalize determine of ResponseType by storing new kind of lambda in #aws_config. 
Is it ok ?

Reply:
> This will retry it silently but with will lose the ability to even log what went wrong

You may write your own retry function and log reason there :)

Reply:
IMO. this should top level application code smth like:
```erlang
my_lambda_retry(AwsReq) when <http 200 but lambda fail> ->
    %% decide if need retry my lambda, do logs and whatever else
    {retry, Req};
  (_) ->  erlcloud_retry:default_retry(AwsReq).
      
 AwsCfg#aws_config{
        retry = fun my_lambda_retry/1,
        retry_num = N
}
```

Reply:
ignore ^

Reply:
@motobob please have a look on new solution